com.ca.lsp.cobol/lsp-core-cobol-parser/src/test/java/com/ca/lsp/core/cobol/visitor/VisitorOutlineAnalysisTest.java

```diff
@@ -0,0 +1,175 @@
+package com.ca.lsp.core.cobol.visitor;
+
+import com.broadcom.lsp.cdi.EngineModule;
+import com.broadcom.lsp.cdi.LangServerCtx;
+import com.broadcom.lsp.cdi.module.databus.DatabusModule;
+import com.broadcom.lsp.domain.cobol.databus.api.DataBusBroker;
+import com.broadcom.lsp.domain.cobol.databus.model.CopybookStorable;
+import com.ca.lsp.core.cobol.engine.CobolLanguageEngine;
+import com.ca.lsp.core.cobol.model.ResultWithErrors;
+import com.ca.lsp.core.cobol.semantics.SemanticContext;
+import com.ca.lsp.core.cobol.semantics.outline.NodeType;
+import com.ca.lsp.core.cobol.semantics.outline.OutlineNode;
+import com.ca.lsp.core.cobol.semantics.outline.OutlineTree;
+import com.google.common.io.Resources;
+import edu.emory.mathcs.backport.java.util.Arrays;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static com.ca.lsp.core.cobol.semantics.outline.NodeType.*;
+import static org.junit.Assert.assertEquals;
+
+public class VisitorOutlineAnalysisTest {
+
+  @BeforeClass
+  public static void setUp() {
+    LangServerCtx.getGuiceCtx(new EngineModule(), new DatabusModule());
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    LangServerCtx.shutdown();
+  }
+
+  private OutlineTree extractOutlineTree(String srcName) throws IOException {
+    DataBusBroker dataBus = LangServerCtx.getInjector().getInstance(DataBusBroker.class);
+    for (String name: List.of("FOO", "BAR", "BAZ")) {
+      dataBus.storeData(CopybookStorable.builder()
+          .name(name)
+          .content("")
+          .uri(name)
+          .build());
+    }
+    CobolLanguageEngine engine = LangServerCtx.getInjector().getInstance(CobolLanguageEngine.class);
+    String program = Resources.toString(Resources.getResource(srcName), Charset.defaultCharset());
```

Comment:
which is the benefit to use the ```google.io``` library instead of use the built-in NIO?

Reply:
I could use `Files.readString` from NIO, instead of `Resources.toString`, but building a path is a nightmare. The possible result could be:
```
String program = Files.readString(Paths.get(getClass().getClassLoader().getResource(srcName).toURI()));
```
And I also need to care about `URISyntaxException`. That's why I decided to use Google library.

Reply:
I got your point and I can accept your impl. Minor thing - bare in mind that ```Resources``` class is in ```@Beta``` according with doc, probably is not a problem for us!
```java
@Beta
@GwtIncompatible
public final class Resources {
//...
}
```

Reply:
Yeah, you are right. But it since the very beginning and still here... OK than.